File: C_Training/Daily_Test/DAY5_2.c
/*Write a C program to decode the given string.

Input Format:

Input consists of String

Output Format:

Output consists ofString

Sample Input:

i2n3s4t3a2

Sample Output:

iinnnsssstttaa*/
#include <stdio.h>
#include <ctype.h> 

void decodeString(const char* str) {
    for (int i = 0; str[i] != '\0'; ++i) {
        if (isdigit(str[i])) {
            int count = str[i] - '0'; 
           
            for (int j = 0; j < count; ++j) {
                putchar(str[i - 1]);
            }
        }
    }
}

int main() {
    char input[100]; 
  
    scanf("%s", input);
 
    decodeString(input);
    printf("\n");

    return 0;
}

File: Java_Training/Daily Tests/DAY5_2.java
/*Gowri is doing her homework. She needs to write a paragraph about Modern History. During that time, she noticed that some words are get repeated again and again. She started counting the number of times a particular word gets repeated. Write a java program to get a string from the user. Count the number of times a word gets repeated in the string.



Input Format:

The first line consists of a string

The second line consists of a string



Output Format:

Refer sample output for formatting specifications



Sample Input:

I felt happy because I saw the others were happy and because I knew I should feel happy

happy



Sample Output:

3

 */


import java.util.Scanner;

public class DAY5_2{
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String inputString = scanner.nextLine().toLowerCase();
        String wordToCount = scanner.nextLine().toLowerCase();
        int count = countOccurrences(inputString, wordToCount);

        System.out.println(count);

        scanner.close();
    }

    private static int countOccurrences(String inputString, String wordToCount) {

        String[] words = inputString.split("\\s+");

        int count = 0;

        for (String word : words) {

            word = word.replaceAll("[^a-zA-Z]", "");

            if (word.equals(wordToCount)) {
                count++;
            }
        }

        return count;
    }
}